client/src/app/s-degree-plan-editor/s-degree-plan-editor.component.ts

```diff
@@ -315,11 +325,16 @@ export class SDegreePlanEditorComponent implements OnInit {
    * @param yearIndex The
    */
   onClickAddNewSemester(yearIndex: number): void {
-    this.planService.addNewSemester(
-      this.plan.years[yearIndex].newSemesterWidget.termSelect,
-      yearIndex,
-      this.plan.years
-    );
+    if (
```

Comment:
Added if statement to create a semester only when the select menu has been set to a valid option.